.github/workflows/build.yml

```diff
@@ -2,36 +2,63 @@ name: Build
 on:
   push:
     branches:
-      - master
+    - dev
   pull_request:
+    branches:
+    - dev
     types: [opened, synchronize, reopened]
+  release:
+    types: [created]
 jobs:
-  sonarcloud:
-    name: SonarCloud
-    runs-on: ubuntu-latest
+  test:
+    name: ${{ matrix.os }} ${{ matrix.gcrypt }} 
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os: ["ubuntu-latest", "ubuntu-18.04", "macOS-latest", "macos-11"]
+        gcrypt: ["--disable-gcrypt", ""]
+
     steps:
-      - uses: actions/checkout@v2
-        with:
-          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
-      - name: Install dependencies
-        run: sudo apt-get install build-essential libc6-dev libpcap-dev libjson-c-dev libgcrypt20-dev gcc-arm-linux-gnueabihf gcc-mingw-w64
-      - name: Configure
-        run: env CFLAGS='-Werror' ./autogen.sh
-      - name: Build
-        run: make all
-      - name: Test DIFF
-        run: ./tests/do.sh
-      - name: Test UNIT
-        run: ./tests/unit/unit
-      - name: Configure (with debug logs)
-        run: make distclean && env CFLAGS='-Werror' ./autogen.sh --enable-debug-messages
-      - name: Build
-        run: make all
-      - name: Configure ARM
-        run: make distclean && env CFLAGS='-Werror' ./autogen.sh --host=arm-linux-gnueabihf --with-only-libndpi
-      - name: Build ARM
-        run: make all
-      - name: Configure Mingw-w64
-        run: make distclean && ./autogen.sh --host=x86_64-w64-mingw32
-      - name: Build Mingw-w64
-        run: make all
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+    - name: Install Ubuntu Prerequisites
+      if: startsWith(matrix.os, 'ubuntu')
+      run: |
```

Comment:
On last question: Is it possible to switch from multi-line statements to single-line statements?

Because with that it should be possible to see exactly which command failed.

(GitLab does support such superfluous things) :grinning:

Reply:
@lnslbrty Done!